[avmfritz] java.lang.NullPointerException: null 	at org.openhab.binding.avmfritz.handler.BoxHandler.updateThingFromDevice

Hi,
I'm using the avmfritz addon with openHAB 2.2.0 Build #1099, Fritz Box 7490 (firmware 6.60),  FRITZ!Powerline 546E and FRITZ!DECT 200. I've added the things via Paper UI. In the log, I get the following error:

>  2017-12-01 17:30:12.122 [DEBUG] [.binding.avmfritz.handler.BoxHandler] - polling FRITZ!Box avmfritz:fritzbox:192_168_178_1
> 2017-12-01 17:30:12.148 [DEBUG] [ternal.hardware.FritzahaWebInterface] - GETting URL http://192.168.178.1/webservices/homeautoswitch.lua?switchcmd=getdevicelistinfos&sid=fb9be0e7973d1f91
> 2017-12-01 17:30:14.120 [DEBUG] [nal.hardware.FritzahaContentExchange] - HTTP response 200
> 2017-12-01 17:30:14.146 [DEBUG] [nal.hardware.FritzahaContentExchange] - HTTP response 200
> 2017-12-01 17:30:14.165 [DEBUG] [nal.hardware.FritzahaContentExchange] - response complete: <devicelist version="1"><device identifier="5C:49:79:ED:20:CF" id="20000" functionbitmask="640" fwversion="06.50" manufacturer="AVM" productname="FRITZ!Powerline 546E"><present>1</present><name>FRITZ!Powerline 546E</name><switch><state>1</state><mode>manuell</mode><lock>0</lock></switch><powermeter><power>72310</power><energy>3264</energy></powermeter></device><device identifier="08761 0300664" id="16" functionbitmask="896" fwversion="03.59" manufacturer="AVM" productname="FRITZ!DECT 200"><present>1</present><name>FRITZ!DECT Keller</name><switch><state>1</state><mode>auto</mode><lock>0</lock></switch><powermeter><power>30750</power><energy>42723</energy></powermeter><temperature><celsius>150</celsius><offset>0</offset></temperature></device></devicelist>
> 
> 2017-12-01 17:30:14.284 [DEBUG] [.binding.avmfritz.handler.BoxHandler] - set device model: org.openhab.binding.avmfritz.internal.ahamodel.DeviceModel@196275e[ain=5C:49:79:ED:20:CF,bitmask=640,isDectRepeater=false,isPowermeter=true,isTempSensor=false,isSwitchableOutlet=true,isHeatingThermostat=false,id=20000,manufacturer=AVM,productname=FRITZ!Powerline 546E,fwversion=06.50,present=1,name=FRITZ!Powerline 546E,org.openhab.binding.avmfritz.internal.ahamodel.SwitchModel@93a881[state=1,mode=MANUAL,lock=0,devicelock=<null>],org.openhab.binding.avmfritz.internal.ahamodel.PowerMeterModel@1afae33[power=72.310,energy=3.264],<null>,<null>]
> 2017-12-01 17:30:14.319 [DEBUG] [.binding.avmfritz.handler.BoxHandler] - set device model: org.openhab.binding.avmfritz.internal.ahamodel.DeviceModel@4eaf98[ain=087610300664,bitmask=896,isDectRepeater=false,isPowermeter=true,isTempSensor=true,isSwitchableOutlet=true,isHeatingThermostat=false,id=16,manufacturer=AVM,productname=FRITZ!DECT 200,fwversion=03.59,present=1,name=FRITZ!DECT Keller,org.openhab.binding.avmfritz.internal.ahamodel.SwitchModel@3cdf58[state=1,mode=AUTOMATIC,lock=0,devicelock=<null>],org.openhab.binding.avmfritz.internal.ahamodel.PowerMeterModel@146b8fe[power=30.750,energy=42.723],org.openhab.binding.avmfritz.internal.ahamodel.TemperatureModel@1ba55d3[celsius=15.0,offset=0.0],<null>]
> 2017-12-01 17:30:14.345 [DEBUG] [.binding.avmfritz.handler.BoxHandler] - update thing avmfritz:FRITZ_DECT_200:192_168_178_1:087610300664 with device model: org.openhab.binding.avmfritz.internal.ahamodel.DeviceModel@4eaf98[ain=087610300664,bitmask=896,isDectRepeater=false,isPowermeter=true,isTempSensor=true,isSwitchableOutlet=true,isHeatingThermostat=false,id=16,manufacturer=AVM,productname=FRITZ!DECT 200,fwversion=03.59,present=1,name=FRITZ!DECT Keller,org.openhab.binding.avmfritz.internal.ahamodel.SwitchModel@3cdf58[state=1,mode=AUTOMATIC,lock=0,devicelock=<null>],org.openhab.binding.avmfritz.internal.ahamodel.PowerMeterModel@146b8fe[power=30.750,energy=42.723],org.openhab.binding.avmfritz.internal.ahamodel.TemperatureModel@1ba55d3[celsius=15.0,offset=0.0],<null>]
> 2017-12-01 17:30:14.387 [ERROR] [.binding.avmfritz.handler.BoxHandler] - null
> java.lang.NullPointerException: null
> 	at org.openhab.binding.avmfritz.handler.BoxHandler.updateThingFromDevice(BoxHandler.java:176) [230:org.openhab.binding.avmfritz:2.2.0.201711260007]
> 	at org.openhab.binding.avmfritz.handler.BoxHandler.addDeviceList(BoxHandler.java:144) [230:org.openhab.binding.avmfritz:2.2.0.201711260007]
> 	at org.openhab.binding.avmfritz.internal.hardware.callbacks.FritzAhaUpdateXmlCallback.execute(FritzAhaUpdateXmlCallback.java:64) [230:org.openhab.binding.avmfritz:2.2.0.201711260007]
> 	at org.openhab.binding.avmfritz.internal.hardware.FritzahaContentExchange.onComplete(FritzahaContentExchange.java:71) [230:org.openhab.binding.avmfritz:2.2.0.201711260007]
> 	at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:193) [75:org.eclipse.jetty.client:9.3.22.v20171030]
> 	at org.eclipse.jetty.client.ResponseNotifier.notifyComplete(ResponseNotifier.java:185) [75:org.eclipse.jetty.client:9.3.22.v20171030]
> 	at org.eclipse.jetty.client.HttpReceiver.terminateResponse(HttpReceiver.java:458) [75:org.eclipse.jetty.client:9.3.22.v20171030]
> 	at org.eclipse.jetty.client.HttpReceiver.responseSuccess(HttpReceiver.java:405) [75:org.eclipse.jetty.client:9.3.22.v20171030]
> 	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.messageComplete(HttpReceiverOverHTTP.java:277) [75:org.eclipse.jetty.client:9.3.22.v20171030]
> 	at org.eclipse.jetty.http.HttpParser.parseContent(HttpParser.java:1617) [78:org.eclipse.jetty.http:9.3.22.v20171030]
> 	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1350) [78:org.eclipse.jetty.http:9.3.22.v20171030]
> 	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.parse(HttpReceiverOverHTTP.java:159) [75:org.eclipse.jetty.client:9.3.22.v20171030]
> 	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.process(HttpReceiverOverHTTP.java:120) [75:org.eclipse.jetty.client:9.3.22.v20171030]
> 	at org.eclipse.jetty.client.http.HttpReceiverOverHTTP.receive(HttpReceiverOverHTTP.java:70) [75:org.eclipse.jetty.client:9.3.22.v20171030]
> 	at org.eclipse.jetty.client.http.HttpChannelOverHTTP.receive(HttpChannelOverHTTP.java:90) [75:org.eclipse.jetty.client:9.3.22.v20171030]
> 	at org.eclipse.jetty.client.http.HttpConnectionOverHTTP.onFillable(HttpConnectionOverHTTP.java:115) [75:org.eclipse.jetty.client:9.3.22.v20171030]
> 	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:283) [79:org.eclipse.jetty.io:9.3.22.v20171030]
> 	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:108) [79:org.eclipse.jetty.io:9.3.22.v20171030]
> 	at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93) [79:org.eclipse.jetty.io:9.3.22.v20171030]
> 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:303) [90:org.eclipse.jetty.util:9.3.22.v20171030]
> 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148) [90:org.eclipse.jetty.util:9.3.22.v20171030]
> 	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136) [90:org.eclipse.jetty.util:9.3.22.v20171030]
> 	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:671) [90:org.eclipse.jetty.util:9.3.22.v20171030]
> 	at org.eclipse.jetty.util.thread.QueuedThreadPool$2.run(QueuedThreadPool.java:589) [90:org.eclipse.jetty.util:9.3.22.v20171030]
> 	at java.lang.Thread.run(Thread.java:748) [?:?]

Any ideas?

--- a/addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/BoxHandler.java
+++ b/addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/BoxHandler.java
@@ -170,25 +170,27 @@ public class BoxHandler extends BaseBridgeHandler implements IFritzHandler {
             }
             if (device.isSwitchableOutlet() && device.getSwitch() != null) {
                 updateThingChannelState(thing, CHANNEL_MODE, new StringType(device.getSwitch().getMode()));
-                updateThingChannelState(thing, CHANNEL_LOCKED, device.getSwitch().getLock().equals(BigDecimal.ONE)
-                        ? OpenClosedType.CLOSED : OpenClosedType.OPEN);
+                updateThingChannelState(thing, CHANNEL_LOCKED,
+                        BigDecimal.ZERO.equals(device.getSwitch().getLock()) ? OpenClosedType.OPEN
+                                : OpenClosedType.CLOSED);
                 updateThingChannelState(thing, CHANNEL_DEVICE_LOCKED,
-                        device.getSwitch().getDevicelock().equals(BigDecimal.ONE) ? OpenClosedType.CLOSED
-                                : OpenClosedType.OPEN);
+                        BigDecimal.ZERO.equals(device.getSwitch().getDevicelock()) ? OpenClosedType.OPEN
+                                : OpenClosedType.CLOSED);
                 if (device.getSwitch().getState() == null) {
                     updateThingChannelState(thing, CHANNEL_SWITCH, UnDefType.UNDEF);
                 } else {
                     updateThingChannelState(thing, CHANNEL_SWITCH,
-                            device.getSwitch().getState().equals(SwitchModel.ON) ? OnOffType.ON : OnOffType.OFF);
+                            SwitchModel.ON.equals(device.getSwitch().getState()) ? OnOffType.ON : OnOffType.OFF);
                 }
             }
             if (device.isHeatingThermostat() && device.getHkr() != null) {
                 updateThingChannelState(thing, CHANNEL_MODE, new StringType(device.getHkr().getMode()));
                 updateThingChannelState(thing, CHANNEL_LOCKED,
-                        device.getHkr().getLock().equals(BigDecimal.ONE) ? OpenClosedType.CLOSED : OpenClosedType.OPEN);
+                        BigDecimal.ZERO.equals(device.getHkr().getLock()) ? OpenClosedType.OPEN
+                                : OpenClosedType.CLOSED);
                 updateThingChannelState(thing, CHANNEL_DEVICE_LOCKED,
-                        device.getHkr().getDevicelock().equals(BigDecimal.ONE) ? OpenClosedType.CLOSED
-                                : OpenClosedType.OPEN);
+                        BigDecimal.ZERO.equals(device.getHkr().getDevicelock()) ? OpenClosedType.OPEN
+                                : OpenClosedType.CLOSED);
                 updateThingChannelState(thing, CHANNEL_ACTUALTEMP,
                         new DecimalType(HeatingModel.toCelsius(device.getHkr().getTist())));
                 updateThingChannelState(thing, CHANNEL_SETTEMP,
@@ -218,7 +220,7 @@ public class BoxHandler extends BaseBridgeHandler implements IFritzHandler {
                     updateThingChannelState(thing, CHANNEL_BATTERY, UnDefType.UNDEF);
                 } else {
                     updateThingChannelState(thing, CHANNEL_BATTERY,
-                            device.getHkr().getBatterylow().equals(HeatingModel.BATTERY_ON) ? OnOffType.ON
+                            HeatingModel.BATTERY_ON.equals(device.getHkr().getBatterylow()) ? OnOffType.ON
                                     : OnOffType.OFF);
                 }
             }

--- a/addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/DeviceHandler.java
+++ b/addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/DeviceHandler.java
@@ -211,27 +211,27 @@ public class DeviceHandler extends BaseThingHandler implements IFritzHandler {
                 break;
             case CHANNEL_RADIATOR_MODE:
                 if (command instanceof StringType) {
-                    if (command.equals(MODE_ON)) {
+                    if (MODE_ON.equals(command)) {
                         state.getHkr().setTsoll(HeatingModel.TEMP_FRITZ_ON);
                         fritzBox.setSetTemp(ain, HeatingModel.TEMP_FRITZ_ON);
                         updateState(CHANNEL_SETTEMP,
                                 new DecimalType(HeatingModel.toCelsius(HeatingModel.TEMP_FRITZ_ON)));
-                    } else if (command.equals(MODE_OFF)) {
+                    } else if (MODE_OFF.equals(command)) {
                         state.getHkr().setTsoll(HeatingModel.TEMP_FRITZ_OFF);
                         fritzBox.setSetTemp(ain, HeatingModel.TEMP_FRITZ_OFF);
                         updateState(CHANNEL_SETTEMP,
                                 new DecimalType(HeatingModel.toCelsius(HeatingModel.TEMP_FRITZ_OFF)));
-                    } else if (command.equals(MODE_COMFORT)) {
+                    } else if (MODE_COMFORT.equals(command)) {
                         BigDecimal comfort_temp = state.getHkr().getKomfort();
                         state.getHkr().setTsoll(comfort_temp);
                         fritzBox.setSetTemp(ain, comfort_temp);
                         updateState(CHANNEL_SETTEMP, new DecimalType(HeatingModel.toCelsius(comfort_temp)));
-                    } else if (command.equals(MODE_ECO)) {
+                    } else if (MODE_ECO.equals(command)) {
                         BigDecimal eco_temp = state.getHkr().getAbsenk();
                         state.getHkr().setTsoll(eco_temp);
                         fritzBox.setSetTemp(ain, eco_temp);
                         updateState(CHANNEL_SETTEMP, new DecimalType(HeatingModel.toCelsius(eco_temp)));
-                    } else if (command.equals(MODE_BOOST)) {
+                    } else if (MODE_BOOST.equals(command)) {
                         state.getHkr().setTsoll(HeatingModel.TEMP_FRITZ_MAX);
                         fritzBox.setSetTemp(ain, HeatingModel.TEMP_FRITZ_MAX);
                         updateState(CHANNEL_SETTEMP,
@@ -309,25 +309,26 @@ public class DeviceHandler extends BaseThingHandler implements IFritzHandler {
             if (device.isSwitchableOutlet() && device.getSwitch() != null) {
                 updateThingChannelState(thing, CHANNEL_MODE, new StringType(device.getSwitch().getMode()));
                 updateThingChannelState(thing, CHANNEL_LOCKED,
-                        device.getSwitch().getLock().equals(BigDecimal.ONE) ? OpenClosedType.CLOSED
-                                : OpenClosedType.OPEN);
+                        BigDecimal.ZERO.equals(device.getSwitch().getLock()) ? OpenClosedType.OPEN
+                                : OpenClosedType.CLOSED);
                 updateThingChannelState(thing, CHANNEL_DEVICE_LOCKED,
-                        device.getSwitch().getDevicelock().equals(BigDecimal.ONE) ? OpenClosedType.CLOSED
-                                : OpenClosedType.OPEN);
+                        BigDecimal.ZERO.equals(device.getSwitch().getDevicelock()) ? OpenClosedType.OPEN
+                                : OpenClosedType.CLOSED);
                 if (device.getSwitch().getState() == null) {
                     updateThingChannelState(thing, CHANNEL_SWITCH, UnDefType.UNDEF);
                 } else {
                     updateThingChannelState(thing, CHANNEL_SWITCH,
-                            device.getSwitch().getState().equals(SwitchModel.ON) ? OnOffType.ON : OnOffType.OFF);
+                            SwitchModel.ON.equals(device.getSwitch().getState()) ? OnOffType.ON : OnOffType.OFF);
                 }
             }
             if (device.isHeatingThermostat() && device.getHkr() != null) {
                 updateThingChannelState(thing, CHANNEL_MODE, new StringType(device.getHkr().getMode()));
                 updateThingChannelState(thing, CHANNEL_LOCKED,
-                        device.getHkr().getLock().equals(BigDecimal.ONE) ? OpenClosedType.CLOSED : OpenClosedType.OPEN);
+                        BigDecimal.ZERO.equals(device.getHkr().getLock()) ? OpenClosedType.OPEN
+                                : OpenClosedType.CLOSED);
                 updateThingChannelState(thing, CHANNEL_DEVICE_LOCKED,
-                        device.getHkr().getDevicelock().equals(BigDecimal.ONE) ? OpenClosedType.CLOSED
-                                : OpenClosedType.OPEN);
+                        BigDecimal.ZERO.equals(device.getHkr().getDevicelock()) ? OpenClosedType.OPEN
+                                : OpenClosedType.CLOSED);
                 updateThingChannelState(thing, CHANNEL_ACTUALTEMP,
                         new DecimalType(HeatingModel.toCelsius(device.getHkr().getTist())));
                 updateThingChannelState(thing, CHANNEL_SETTEMP,
@@ -357,7 +358,7 @@ public class DeviceHandler extends BaseThingHandler implements IFritzHandler {
                     updateThingChannelState(thing, CHANNEL_BATTERY, UnDefType.UNDEF);
                 } else {
                     updateThingChannelState(thing, CHANNEL_BATTERY,
-                            device.getHkr().getBatterylow().equals(HeatingModel.BATTERY_ON) ? OnOffType.ON
+                            HeatingModel.BATTERY_ON.equals(device.getHkr().getBatterylow()) ? OnOffType.ON
                                     : OnOffType.OFF);
                 }
             }

